src/layers/basic.jl

```diff
@@ -191,8 +191,8 @@ Dense(W::LinearAlgebra.Diagonal, bias = true, σ = identity) =
   Scale(W.diag, bias, σ)
 
 function _size_check(layer, x::AbstractArray, (d, n)::Pair)
-  d > 0 || throw(DimensionMismatch(string("layer ", layer,
-    " expects ndims(input) > ", ndims(x)-d, ", but got ", summary(x))))
+  d <= ndims(x) || throw(DimensionMismatch(string("layer ", layer,
```

Comment:
To protect against typos, I'd still check if `d` is bigger than `0`.
```suggestion
  0 < d <= ndims(x) || throw(DimensionMismatch(string("layer ", layer,
```